### ISSUE TITLE
Ability to create more types of descriptors

### ISSUE THREAD

**Issue by BitcoinZavior** [LtbLightning/bdk-flutter#5]
Large variety of descriptors are possible reference: https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md

`createDescriptor` should be able to create as many of these descriptors as possible.

**Comment by i5hi** [comment]
We would need to add two things for this:

- policy compiler 
- newBip84Sh() method to the Descriptor class to support `wsh` script type

**Comment by i5hi** [comment]
This would then also require the `policy_id` function to be exposed. It would up the user to know what they are doing if they venture into creating complex scripts and they would have to know how to parse their policy and specify which nodes in the policy they are signing for.

**Comment by BitcoinZavior** [comment]
Yes, we will probably follow bdk-ffi api which in turn follows bdk api.
